src/ray/util/event.cc

```diff
@@ -57,10 +57,22 @@ LogEventReporter::~LogEventReporter() { Flush(); }
 
 void LogEventReporter::Flush() { log_sink_->flush(); }
 
-std::string LogEventReporter::EventToString(const rpc::Event &event) {
-  std::stringstream result;
+std::string LogEventReporter::replaceLineFeed(std::string message) {
+  std::stringstream ss;
+  // If the message has \n or \r, we will replace with \\n
+  for (int i = 0, len = message.size(); i < len; ++i) {
+    if (message[i] == '\n' || message[i] == '\r') {
+      ss << "\\n";
+    } else {
+      ss << message[i];
+    }
+  }
+  return ss.str();
+}
 
-  boost::property_tree::ptree pt;
+std::string LogEventReporter::EventToString(const rpc::Event &event,
```

Comment:
Btw, do we need this? Why don't we just store the json serialized string instead of copying to the protobuf first and re-copy to the json string?

Reply:
Yep, this is a problem of history. In the first version of event framework, the event could be transform directly by RPC. But now, we just write an event log. We should remove the protobuf in future. I have added an TODO https://github.com/ray-project/ray/pull/17476/files#diff-01dea7b9807ba283ccf8ffd3131551fd30e0b6d74402f8b07253e256a5b28469R96.